invoice_discount/model/invoice.py

```diff
@@ -23,133 +23,65 @@
-        invoice lines.
-        @param self: The object pointer.
-        @param cr: A database cursor
-        @param uid: ID of the user currently logged in
-        @param ids: list of ids for which name should be read
-        @param field_name: field that call the method
-        @param arg: Extra arguments
-        @param context: A standard dictionary
-        @return : Dict with values
-        """
-        context = context or {}
-        total = 0.0
-        res = {}
-        for inv in self.browse(cr, uid, ids, context=context):
-            for line in inv.invoice_line:
-                total += line.subtotal_wo_discount
-            res[inv.id] = total
-        return res
-
-    def _get_discount(self, cr, uid, ids, args, field_name, context=None):
-        """Method to get the amount total of discount in the invoice lines.
-        @param self: The object pointer.
-        @param cr: A database cursor
-        @param uid: ID of the user currently logged in
-        @param ids: list of ids for which name should be read
-        @param field_name: field that call the method
-        @param arg: Extra arguments
-        @param context: A standard dictionary
-        @return : Dict with values
-        """
-        context = context or {}
-        total = 0.0
-        res = {}
-        for inv in self.browse(cr, uid, ids, context=context):
-            for line in inv.invoice_line:
-                total += line.discount_amount
-            res[inv.id] = total
-        return res
+    subtotal_wo_discount = fields.Float(
+        compute='_compute_subtotal_wo_disc', string='SubTotal w/o Discount',
+        store=True,
+        help='Amount without apply the discount of the lines of the invoice.')
+    discount_amount = fields.Float(
+        compute='_compute_subtotal_wo_disc',  string='Discount', store=True,
+        help='Total of discount apply in each line of the invoice.')
 
-    _columns = {
-        'subtotal_wo_discount': fields.function(_get_subtotal_without_discount,
-                                                string='SubTotal w/o Discount',
-                                                store=True, type='float',
-                                                help='Amount without apply the\
-                                                 discount of the lines of the \
-                                                invoice.'),
-        'discount_amount': fields.function(_get_discount, string='Discount',
-                                           store=False, type='float',
-                                           help='Total of discount apply in \
-                                           each line of the invoice.'),
-    }
+    @api.depends('invoice_line.subtotal_wo_discount',
+                 'invoice_line.discount_amount')
+    def _compute_subtotal_wo_disc(self):
+        # TODO: Use read_group method here
+        for invoice in self:
```

Comment:
@josemoralesp I believe that code wouldn't work because of two reasons:
- When an invoice is being modified, its ID is not accessible, it's a NewId object. We could workaround that using `invoice._cache['id']`
- If we use `groupby='invoice_id'`, then there would be only one result, because all lines belongs to the same invoice.

Am I wrong?